lua/lualine/init.lua

```diff
@@ -266,10 +266,22 @@ end
 local function tabline() return statusline(config.tabline, true) end
 
 local function setup_theme()
-  if type(config.options.theme) == 'string' then
-    config.options.theme = require('lualine.themes.' .. config.options.theme)
+  local theme_name = config.options.theme
+  local theme
+  if type(theme_name) == 'string' then
+    local ok
+    ok, theme = pcall(require, 'lualine.themes.' .. theme_name)
```

Comment:
this is weird... Wont' having `local ok, theme = ...` the same effect?

Reply:
Theme is required out of the if scope . If i do `local ok, theme = ..` it will create a new var named theme and shadow the one from outer scope .

Reply:
```lua
  if type(theme_name) == 'string' then
    local ok, theme = pcall(require, 'lualine.themes.' .. theme_name)
    if ok then
      highlight.create_highlight_groups(theme)
    else
      vim.api.nvim_echo({
        {
          'theme ' .. theme_name .. ' not found defaulting to gruvbox',
          'ErrorMsg'
        }
      }, true, {})
      local gruvbox = require 'lualine.themes.gruvbox'
      highlight.create_highlight_groups(gruvbox)
    end
  end
```

Reply:
I want to allow custon themes to be directly passed though theme option  . Currently it's easily posible to require a theme file change some colors and use that option theme to that table and have it work . Your implementation will break it . as table themes aren't used

Reply:
Will this work for you ?

```lua
local ok, theme
  if type(theme_name) == 'string' then
    ok, theme = pcall(require, 'lualine.themes.' .. theme_name)
```

Reply:
oh right, I see what you mean. Your latest suggestion is the best one so we can stick with it

Reply:
Ok I've made the changes.